### PR TITLE
Add vector-vector * methods for Transpose / Adjoint

### PR DESCRIPTION
**Pull request by glennmoy**

### CODE REVIEW

File: src/functions_math.jl [hunk 1]
```diff
@@ -54,10 +54,14 @@ for A in (
     Adjoint{<:Number,<:AbstractVector},
     Transpose{<:Real,<:AbstractVector{<:Real}},
     Transpose{<:Any, <:AbstractMatrix{T}} where T,  # resolves ambiguity error in AxisKeys
+    Union{Adjoint{T, <:AbstractVector{T}}, Transpose{T, <:AbstractVector{T}}} where {T},
```
Comment (mzgubic) [inline src/functions_math.jl:57]: why not just
```suggestion
    Adjoint{T, <:AbstractVector{T}} where {T},
    Transpose{T, <:AbstractVector{T}}} where {T},
```

File: src/functions_math.jl [hunk 2]
```diff
@@ -54,10 +54,14 @@ for A in (
...
 )
     @eval function Base.:*(a::$A, b::NamedDimsArray{L,T,1,<:AbstractVector{T}}) where {L,T}
         return *(a, parent(b))
     end
+    @eval function Base.:*(b::NamedDimsArray{L,T,1,<:AbstractVector{T}}, a::$A) where {L,T}
```
Comment (mzgubic) [inline src/functions_math.jl:62]: Do we need this one? I think the `CoVector` one above is meant to take care of these.

**Review by mzgubic — COMMENTED** [review]
Should we also add tests?

File: src/functions_math.jl (hunk 1 above)
Comment (glennmoy) [reply src/functions_math.jl:57]: updated

**Pull request merged**